### ISSUE TITLE
Old save games are broken

### ISSUE THREAD

**Issue by hweimer** [raceintospace/raceintospace#875]
I just tried to load some older save games, including some using the legacy format and some using the newer JSON-based ones. They newer ones failed complaining about a missing Race field and the old ones crashed right away (I assume the internal memory race layout got changed when adding the race variable to the naut structures).

**Comment by peyre** [comment]
Yes, I lost most of my save games as well.  I just figured it was the way it was now and moved on.

**Comment by Alsed** [comment]
I've been reading old issues (open and closed) and, If I understand correctly, several decisions were made, namely about maintaining data structures and adding libraries (such as protobuff), to preserve backward compatibility with previous versions of the save files. If this is no longer the case, it would involve several changes.
So, this seems to be an important discussion. If we still intend to preserve the stability of the old save files, then this is a high priority issue. 
If not, then this would not an issue, _but_ the decision about the file save structure (and other data structures) is reopened.

**Comment by hweimer** [comment]
Protobuf is not being used at the moment. The game serializes data structures via the cereal library. There is supposed to be some code to load save games in the binary (pre-cereal) format, but the internal data structures have diverged from the save files when the race category was added.

One way to address this issue would be to keep a struct for legacy data, serialize to JSON from that and then deserialize into the actual game data struct.
